host/host.go

```diff
@@ -68,24 +68,79 @@ type (
 		Cap() uint64
 	}
 
-	// A ContractStore stores contracts, metadata, and proofs for the host.
+	// A SettingsReporter returns the host's current settings.
+	SettingsReporter interface {
+		Settings() rhp.HostSettings
+	}
+
+	// A TransactionPool broadcasts transaction sets to miners for inclusion in
+	// an upcoming block.
+	TransactionPool interface {
+		AddTransaction(txn types.Transaction) error
+		RecommendedFee() types.Currency
+		Transactions() []types.Transaction
+	}
+
+	// A Wallet provides addresses and funds and signs transactions.
+	Wallet interface {
+		Address() types.Address
+		FundTransaction(txn *types.Transaction, amount types.Currency, pool []types.Transaction) ([]types.ElementID, func(), error)
+		SignTransaction(vc consensus.ValidationContext, txn *types.Transaction, toSign []types.ElementID) error
+	}
+)
+
+type (
+	// ContractState is the current lifecycle stage of a contract.
+	ContractState string
+
+	// A Contract contains metadata on the current lifecycle stage of a file
+	// contract.
+	Contract struct {
+		rhp.Contract
+		Parent types.FileContractElement
```

Comment:
perhaps these fields could be replaced with `Revision types.FileContractRevision`? Unless there's value in storing it as an `rhp.Contract`.